### PR TITLE
Polling added for windows

### PR DESCRIPTION
**Pull request by Contrevien**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -81,20 +86,40 @@ class NetworkStateNotifier extends React.Component {
...
+    );
+  };
+
+  handleChecker = () => {
+    const { polling, isOnline } = this.state;
+
+    if (polling) {
+      fetch(pollingURL, {}, 2000)
+        .then((res) => { // eslint-disable-line
+          if (!isOnline) {
```
Comment (Arjun-sna) [inline src/index.js]: Don't you need to check the status of the response?

File: src/index.js [hunk 2]
```diff
@@ -81,20 +86,40 @@ class NetworkStateNotifier extends React.Component {
     });
   };
 
-  handleChecker = () => {
-    const { isOnline } = this.state;
+  toggleOnlineOffline = (status) => {
```
Comment (Arjun-sna) [inline src/index.js]: this function sets the online status based `status` parameter. so it should be named with `set` prefix. It is not just toggling always.

File: src/index.js (hunk 1 above)
Comment (Contrevien) [reply src/index.js]: I believe as long as there is a response wouldn't that mean that host is connected to the internet?

File: src/index.js [hunk 3]
```diff
@@ -3,16 +3,21 @@ import PropTypes from 'prop-types';
...
-      messages: [],
-    };
-  }
+class NetworkStateNotifier extends React.Component {
+  state = {
+    checker: null,
+    isOnline:
+      inBrowser && typeof navigator.onLine === 'boolean'
+        ? navigator.onLine
+        : true,
```
Comment (Arjun-sna) [inline src/index.js]: setting the default value to `true`?

File: src/index.js (hunk 2 above)
Comment (Contrevien) [reply src/index.js]: Ok I will rename it

File: src/index.js (hunk 3 above)
Comment (Contrevien) [reply src/index.js]: This case arises when the user is probably running a script and not in a browser and there no property for navigator object/it has been manipulated. It was a confusing decision.

File: src/index.js [hunk 4]
```diff
@@ -3,16 +3,21 @@ import PropTypes from 'prop-types';
```
Comment (Arjun-sna) [inline src/index.js]: Just a suggestion

`handleMessageAddition` function can be refactored like below
```
messages.push(
        <Message
          message={isOnline ? onlineMessage : offlineMessage}
          style={{
            ...messageStyles,
            backgroundColor: isOnline ? onlineColor : offlineColor,
          }}
          className={messageClassName}
        />,
      );
```

Comment (Contrevien) [reply src/index.js]: Thanks I will do that!

File: src/index.js (hunk 1 above)
Comment (Arjun-sna) [reply src/index.js]: My bad. Missed that point.

File: src/index.js (hunk 3 above)
Comment (Arjun-sna) [reply src/index.js]: You can set it to undefined. which is again a false value. I don't think setting to true is good.

Comment (Contrevien) [reply src/index.js]: I will set it undefined and run the tests. But wouldn't setting false be the same?

Comment (Arjun-sna) [reply src/index.js]: false is fine too. But the point is you don't the value of isOnline exactly. So I suggested undefined.

**Pull request merged**